testsuite/tests/kuadrant/limitador/test_route_rule_child.py

```diff
@@ -0,0 +1,36 @@
+"""Tests that RLP for a child of HTTPRouteRule doesn't take effect if there isn't a matching HTTPRouteRule"""
+
+import pytest
+
+from testsuite.gateway import RouteMatch, PathMatch, MatchType, HTTPMethod
+from testsuite.policy.rate_limit_policy import Limit, RouteSelector
+
+
+@pytest.fixture(scope="module")
+def route(route, backend):
+    """Add new rule to the route"""
+    route.remove_all_rules()
+    route.add_rule(
+        backend,
+        RouteMatch(path=PathMatch(value="/anything", type=MatchType.PATH_PREFIX), method=HTTPMethod.GET),
+    )
+    return route
+
+
+@pytest.fixture(scope="module")
+def rate_limit(rate_limit):
+    """Add limit to the policy"""
+    selector = RouteSelector(
+        RouteMatch(path=PathMatch(value="/anything/get", type=MatchType.EXACT), method=HTTPMethod.GET)
```

Comment:
This is not a child but a subset, it is the more concrete rule. Does it in this case limit if /anything/get is used? If yes, we should test that it correctly limits the subset, if no, I do not see how it is different from any other invalid reference to a rule

Reply:
It doesn't limit `anything/get`, but what makes it different from an invalid refence would be the `/anything` PathPrefix route rule covering the subset we want to bind our rate policy to despite there not being a specific rule to bind to.

Reply:
So I set up rule to "/anything/get" and it does not limit "/anything/get"?

Reply:
yes, in this configuration `/anything/get` should not be limited because there isn't an exact matching route rule for the RLP, only a route rule matching the paths with prefix `/anything`